examples/js/exporters/GLTFExporter.js

```diff
@@ -1747,99 +1767,191 @@ THREE.GLTFExporter.prototype = {
-						var totalByteLength = GLB_HEADER_BYTES
-							+ jsonChunkPrefix.byteLength + jsonChunk.byteLength
-							+ binaryChunkPrefix.byteLength + binaryChunk.byteLength;
-						headerView.setUint32( 8, totalByteLength, true );
-
-						var glbBlob = new Blob( [
-							header,
-							jsonChunkPrefix,
-							jsonChunk,
-							binaryChunkPrefix,
-							binaryChunk
-						], { type: 'application/octet-stream' } );
-
-						var glbReader = new window.FileReader();
-						glbReader.readAsArrayBuffer( glbBlob );
-						glbReader.onloadend = function () {
-
-							onDone( glbReader.result );
+				reader.readAsArrayBuffer( blob );
+				reader.onloadend = function () {
 
-						};
+					var binaryChunk = getPaddedArrayBuffer( reader.result );
+					resolve( binaryChunk );
 
-					};
+				};
 
-				} else {
+				reader.onerror = reject;
+
+			} );
 
-					reader.readAsDataURL( blob );
-					reader.onloadend = function () {
+		}
 
-						var base64data = reader.result;
-						outputJSON.buffers[ 0 ].uri = base64data;
-						onDone( outputJSON );
+		var pending = [];
+		var blobParts = [];
 
-					};
+		// GLB header.
+		var header = new ArrayBuffer( GLB_HEADER_BYTES );
+		var headerView = new DataView( header );
+		headerView.setUint32( 0, GLB_HEADER_MAGIC, true );
+		headerView.setUint32( 4, GLB_VERSION, true );
 
-				}
+		blobParts.push( header );
+
+		// JSON chunk.
+		var jsonChunk = getPaddedArrayBuffer( stringToArrayBuffer( JSON.stringify( chunks.json ) ), 0x20 );
+		var jsonChunkPrefix = new DataView( new ArrayBuffer( GLB_CHUNK_PREFIX_BYTES ) );
+		jsonChunkPrefix.setUint32( 0, jsonChunk.byteLength, true );
+		jsonChunkPrefix.setUint32( 4, GLB_CHUNK_TYPE_JSON, true );
+
+		blobParts.push( jsonChunkPrefix, jsonChunk );
+
+		if ( chunks.buffers.length !== 0 ) {
+
+			var pendingBinChunk = readBinArrayBuffer( chunks.buffers[ 0 ] ).then( function ( binaryChunk ) {
+
+				var binaryChunkPrefix = new DataView( new ArrayBuffer( GLB_CHUNK_PREFIX_BYTES ) );
+				binaryChunkPrefix.setUint32( 0, binaryChunk.byteLength, true );
+				binaryChunkPrefix.setUint32( 4, GLB_CHUNK_TYPE_BIN, true );
+
+				var totalByteLength = GLB_HEADER_BYTES
+					+ jsonChunkPrefix.byteLength + jsonChunk.byteLength
+					+ binaryChunkPrefix.byteLength + binaryChunk.byteLength;
+				headerView.setUint32( 8, totalByteLength, true );
+
+				blobParts.push( binaryChunkPrefix, binaryChunk );
+
+			} );
+
+			pending.push( pendingBinChunk );
+
+		}
+
+		Promise.all( pending ).then( function ( ) {
+
+			var glbBlob = new Blob( blobParts, { type: 'application/octet-stream' } );
+			onDone( glbBlob );
+
+		} ).catch( onError );
+
+	},
+
+	/**
+	 * Parse scenes and generate GLTF output
+	 * @param  {THREE.Scene or [THREE.Scenes]} input   THREE.Scene or Array of THREE.Scenes
+	 * @param  {Function} onDone  Callback on completed
+	 * @param  {Function} onError  Callback on error
+	 * @param  {Object} options options
+	 */
+	parse: function ( input, onDone, onError, options ) {
```

Comment:
For backward-compatibility, could you check if onError is an object then log a warning? A deprecation warning for the removed "binary" option would be good, too.